File: Leetcode 2101-2200/2136/2136.java
// Leetcode 2136: Earliest Possible Day of Full Bloom
// https://leetcode.com/problems/earliest-possible-day-of-full-bloom/
// Solved on 11th of December, 2025
import java.util.Arrays;

class Solution {
    private record SeedInfo(int plantTime, int growTime) {}

    /**
     * Calculates the earliest possible day all flowers can bloom.
     * @param plantTime An array where plantTime[i] is the time it takes to plant the i-th flower.
     * @param growTime An array where growTime[i] is the time it takes for the i-th flower to grow.
     * @return The earliest possible day all flowers can bloom.
     */
    public int earliestFullBloom(int[] plantTime, int[] growTime) {
        int n = plantTime.length;
        SeedInfo[] seeds = new SeedInfo[n];

        for (int i = 0; i < n; i++) {
            seeds[i] = new SeedInfo(plantTime[i], growTime[i]);
        }

        Arrays.sort(seeds, (a, b) -> b.growTime - a.growTime);

        int plantingDays = 0;
        int earliestBloomDay = 0;

        for (SeedInfo seed : seeds) {
            int finishPlantingDay = plantingDays + seed.plantTime;

            int currentSeedBloomDay = finishPlantingDay + seed.growTime;

            earliestBloomDay = Math.max(earliestBloomDay, currentSeedBloomDay);

            plantingDays = finishPlantingDay;
        }

        return earliestBloomDay;
}

File: Leetcode 2101-2200/2136/2136.py
# Leetcode 2136: Earliest Possible Day of Full Bloom
# https://leetcode.com/problems/earliest-possible-day-of-full-bloom/
# Solved on 11th of December, 2025
class Solution:
    def earliestFullBloom(self, plantTime: list[int], growTime: list[int]) -> int:
        """
        Calculates the earliest possible day all flowers can be in full bloom.

        Args:
            plantTime (list[int]): A list of integers representing the time it takes to plant each flower.
            growTime (list[int]): A list of integers representing the time it takes for each flower to grow after planting.

        Returns:
            int: The earliest day all flowers are in full bloom.
        """
        flowerData = sorted(zip(plantTime, growTime), key=lambda x: x[1], reverse=True)
        currentPlantTime = 0
        maxBloomTime = 0

        for plant, grow in flowerData:
            currentPlantTime += plant
            maxBloomTime = max(maxBloomTime, currentPlantTime + grow)

        return maxBloomTime